cosmos/operators/local.py

```diff
@@ -690,10 +698,21 @@ def register_dataset(self, new_inlets: list[Dataset], new_outlets: list[Dataset]
                 DAG.bulk_write_to_db([self.dag], session=session)  # type: ignore[attr-defined, call-arg, arg-type]
                 session.commit()
         else:
-            logger.info("Assigning inlets/outlets with DatasetAlias")
             dataset_alias_name = get_dataset_alias_name(self.dag, self.task_group, self.task_id)  # type: ignore[attr-defined]
-            for outlet in new_outlets:
-                context["outlet_events"][dataset_alias_name].add(outlet)  # type: ignore[index]
+
+            if AIRFLOW_VERSION.major == 2:
+                logger.info("Assigning inlets/outlets with DatasetAlias in Airflow 2")
+
+                for outlet in new_outlets:
+                    context["outlet_events"][dataset_alias_name].add(outlet)  # type: ignore[index]
+            else:  # AIRFLOW_VERSION.major == 3
```

Comment:
```suggestion
            elif AIRFLOW_VERSION.major == 3
```
I think explicit check is better. I hope no one is using Airflow 1.x anymore :D, but still explicit check for Airflow 3 might be better, and if needed another else block for not 2 & 3 version would be better IMO.